### ISSUE TITLE
addSchema on SchemaRouter should have a Handler or return a future

### ISSUE THREAD

**Issue by LouizFC** [eclipse-vertx/vertx-json-schema#30]
#### Describe the feature

We already have `resolveRef` for querying schemas, but as in the title implies, the `SchemaRouter` needs a asynchronous way to addSchemas to it.

#### Use cases

I have an application that gets gets schemas from various places. These schemas are versioned, so they can be safely cached. I created a SchemaRouter that persists the schemas to the disk. Problem is, in my implementation, `addSchema` can fail. My current workaround is wrapping `addSchema` calls unto a `executeBlocking` Promise.

#### Contribution

I can contribute a small patch if needed. I don't know the full implication of such changes, but since this package is marked as "Technical Preview", such change should be expected by users.

**Comment by slinkydeveloper** [comment]
@LouizFC I'm trying to figure out what would you expect as a user. Do you want something like `addSchema` that gets a path and under the hood resolves the schema (from FS or from HTTP)?

```
@Fluent SchemaRouter addSchema(JsonPointer absolutePointer)
```

I don't get how the interface should be async, can you provide an example of using the API you expect?

**Comment by LouizFC** [comment]
@slinkydeveloper Thank you for responding.

I have a `SchemaStore` that implements `SchemaRouter`, it delegates most things to a underlying `SchemaRouterImpl`, it stores the resolved schemas to the disk. It implemented `SchemaRouter` to abuse the fact that `addSchema` and `addSchemaWithScope` is always called by the parser.

I was kinda burned out yesterday, so I don't know exactly what I was thinking when I opened this, I am sorry.

Today I refactored the code to split the storage from the Router responsibilities, `resolveRef` already does what I need for, well, resolving.

Thanks again for the work put on this library. Best regards
